File: kingpyn/test/unit/crcTest.py
'''
Created on Aug 31, 2015

@author: peter
'''
import unittest

from hdlc.crc import crc_ccitt_update

class Test(unittest.TestCase):


    def setUp(self):
        pass


    def tearDown(self):
        pass


    def testCrc_ccitt_update(self):
        crc = 0xFFFF;
        
        crc = crc_ccitt_update(crc, 0x00)
        self.assertTrue(0xE1F0 == crc)
    
        crc = crc_ccitt_update(crc, 0x01)
        self.assertTrue(0x0D2E == crc)
    
        crc = crc_ccitt_update(crc, 0x02)
        self.assertTrue(0xDFEF == crc)
    
        crc = crc_ccitt_update(crc, 0xDF)
        crc = crc_ccitt_update(crc, 0xEF)
        self.assertTrue(0x0000 == crc)


if __name__ == "__main__":
    #import sys;sys.argv = ['', 'Test.testCrc_ccitt_update']
    unittest.main()

File: test/src/CrcTest.cpp
/*
 * CrcTest.cpp
 *
 *  Created on: Sep 1, 2015
 *      Author: peter
 */

#define BOOST_TEST_DYN_LINK
#include <boost/test/unit_test.hpp>

#include <crc.h>

BOOST_AUTO_TEST_CASE( crc_ccitt_update_test ) {
	uint16_t crc = 0xFFFF;

	crc_ccitt_update(crc, 0x00);
	BOOST_CHECK(0xE1F0 == crc);

	crc_ccitt_update(crc, 0x01);
	BOOST_CHECK(0x0D2E == crc);

	crc_ccitt_update(crc, 0x02);
	BOOST_CHECK(0xDFEF == crc);

	crc_ccitt_update(crc, 0xDF);
	crc_ccitt_update(crc, 0xEF);
	BOOST_CHECK(0x0000 == crc);
}
